docs/core/deploying/native-aot/libraries.md

```diff
@@ -0,0 +1,20 @@
+---
+title: Building native libraries
+description: How to build native libraries with Native AOT
+author: agocke
+ms.author: angocke
+ms.date: 04/17/2024
+---
+
+# Building native libraries
+
+Publishing .NET class libraries as Native AOT allows creating libraries that can be consumed from non-.NET programming languages. The produced native library is self-contained and doesn't require a .NET runtime to be installed.
+
+> [!NOTE]
+> Only "shared libraries" (also known as DLLs on Windows) are supported. Static libraries are not officially supported and may require compiling Native AOT from source.
+
+Publishing a class library as Native AOT creates a native library that exposes methods of the class library annotated with <xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute> with a non-null `EntryPoint` field. For more information, see the [native library sample](https://github.com/dotnet/samples/tree/main/core/nativeaot/NativeLibrary) available in the dotnet/samples repository on GitHub.
+
+## Limitations
+
+Event Source relies on process-wide state that prevents multiple copies from running concurrently in a single process. If multiple Native AOT libraries, or a Native AOT library and CoreCLR, could be loaded into the same process, ensure that `<EventSourceSupport>` is set to `false`.
```

Comment:
```suggestion
`EventSource` is not supported or recommended when compiling to a native library. Please go to https://github.com/dotnet/runtime/issues/91762 for more details.
```
This is what [the build error message](https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets#L54) says. Should we just stick with that? In any case, the build warning and this doc should be in sync.

The full story is more nuanced:
- The global state problem is in EventPipe event channel. Other diagnostic event channels like ETW (perfview) or in-proc EventSource listeners should work fine.
- The second in-proc instance of EventPipe fails to initialize silently, but it does not seem to cause any harm.
- The first instance of EventPipe initializes fine, except that it does not close the session properly during shutdown when compiled into a library. It should not be a problem for most diagnostic tooling, but there may be cases here and there where the diagnostic tools choke on it.

I have tried to validate these claims but run into problems with basic things being broken even without libraries in the picture. Opened https://github.com/dotnet/runtime/issues/101328 .

Reply:
I forgot we added a warning here. I think that’s even better than docs as this limitation might just be a point in time issue, and the warning is a more direct feedback mechanism. 

I’ve removed this section entirely. 

We can track fixing these issues separately, prioritized by customer demand.